services/streammanager/kafka/client/producer.go

```diff
@@ -117,5 +126,22 @@ func (p *Producer) Publish(ctx context.Context, msgs ...Message) error {
 			Headers: headers,
 		}
 	}
+
 	return p.writer.WriteMessages(ctx, messages...)
 }
+
+var tempError interface{ Temporary() bool }
+
+func IsProducerErrTemporary(err error) bool {
```

Comment:
> and consider the kafka.WriteErrors as temporary only if all contained errors are temporary themselves.

I'm not sure about this because if only one is temporary then we would return 400 and that message won't be retried leading to data loss, correct?

Reply:
correct, retrying as long as there is at least 1 temporary errors would be better 👍

Reply:
Fix pushed :+1: